### ISSUE TITLE
The script for training catSeqTG-2RF1 freezes with no CPU or GPU utilization

### ISSUE THREAD

**Issue by amarnamarpan** [kenchan0226/keyphrase-generation-rl#19]
Everytime I try to train the catSeqTG-2RF1 model from catSeqTG, the script suddenly stops in between at a random place. The last time, I have waited for a day before force stopping the script. While frozen, the script holds on to the Primary Memory space and the GPU memory space, but no processing or disk is used. I am using PyTorch 0.4 on Windows. 
Is it due to some deadlock? I don't understand.
the following is my output:
until I ctrl+break out of the frozen state.
`OUTPUT:
INFO:root:======================  Model Parameters  =========================
12/06/2020 13:50:17 [INFO] train: ======================  Model Parameters  =========================
INFO:root:Training a seq2seq model with copy mechanism
12/06/2020 13:50:17 [INFO] train: Training a seq2seq model with copy mechanism
C:\Users\Arpan\anaconda3\envs\py36\lib\site-packages\torch\nn\modules\rnn.py:60: UserWarning: dropout option adds dropout after all but last recurrent layer, so non-zero dropout expects num_layers greater than 1, but got dropout=0.1 and num_layers=1
  "num_layers={}".format(dropout, num_layers))
Epoch 1; batch: 0; total batch: 0
INFO:root:Time for training: 863.4
^C
`

**Comment by kenchan0226** [comment]
That’s weird, can you insert some print script in the training loop to see
which part it get struck? I haven’t tested our code in windows, it works in
Linux environment.

**Comment by amarnamarpan** [comment]
I do not understand the working behind the code. But for some reason if i run the code with bath_worker=1 it does not freeze. Anyway, I changed a line (line 75) from **train_rl.py** to print the current batch and total batch more frequently and to check after how many batches does the training get stuck. And it does get randomly stuck. Notably if I am heavily loading the machine with other RAM intensive tasks such as opening many tabs on chrome, it gets stuck earlier. Maybe there's some problem with the windows version of PyTorch. 
Anyway, I can train now with batch_workers=1 for now. Thanks.
